include/rewd_controllers/FTThresholdClient.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef REWD_CONTROLLERS_FTTHRESHOLDCLIENT_HPP
+#define REWD_CONTROLLERS_FTTHRESHOLDCLIENT_HPP
+
+#include <actionlib/client/simple_action_client.h>
+#include <pr_control_msgs/SetForceTorqueThresholdAction.h>
+#include <ros/ros.h>
+
+namespace rewd_controllers {
+
+/// The FTThresholdClient configures the MoveUntilTouch(Topic)Controller's
+/// thresholds.
+/// When those thresholds are exceeded, the controller stops the movement.
+class FTThresholdClient
+{
+
+public:
+  /// Constructor.
+  FTThresholdClient(const std::string& controllerThresholdTopic, ros::NodeHandle nodeHandle);
+
+  /// Sets the MoveUntilTouchControllers Thresholds.
+  /// Blocks until the threshold could be set successfully.
+  /// Can be aborted with !ros::ok(), in which case it returns false
+  bool trySetThresholdRepeatedly(double forceThreshold, double torqueThreshold);
```

Comment:
The semantic difference is that it actually sends the goal again and again, instead of only one time, like in the `trySetThreshold` function.
The reason for this feature is that when we start an Action Server, it doesn't react to goals from the past, so we need to send a fresh one.

One could argue that it'd be better to do every function like this (sending the goal every second), which would also make this method obsolete. What do you think about that?

Reply:
Hmm, my understanding is that this should only be called after the action server has been instantiated (because we call `waitForServer` in the constructor). Is that not the case?

I don't think it's a good idea to make every function like this; we should remain explicit about when we're retrying. I think the default usage should remain to just try once (and return success).

Reply:
Oh, you're right, that's the way it should be happening. I don't remember the reasoning I had when I programmed it like this in `ada_demos`. I'll check if it's still necessary.

Reply:
Alright. I found out the original reason for this. Here we go:
Two years ago, @ClintLiddick wanted to force users to wait until the sensor is calibrated, which takes a while. In this time, the `MoveUntilTouchController` does not accept setting the thresholds. See [this line](https://github.com/personalrobotics/rewd_controllers/blob/master/src/MoveUntilTouchController.cpp#L181).
Because the controller throws away the goal, we set it repeatedly.

So the question is: Keep the repeated sending as it is or refactor the `MoveUntilTouchController`?

Reply:
I see. I think that means that as long as we wait until the sensor is calibrated, we shouldn't need to retry. Is there some other way to block until the sensor is calibrated? (By refactoring, do you mean providing some way to block until it's calibrated?)

To be clear, I'm fine with the repeated sending as is, but it seems that we should eventually have some way to avoid it entirely. If that doesn't exist, let's create an issue to make sure we remember that that's a problem.

Reply:
By refactoring, I mean providing some other way to block the controller usage while the sensor is calibrating, because setting the thresholds and waiting for calibration don't have much to do with each other except that they're both done in the beginning. I have a couple of ideas, but tbh I don't like either of them very much.

If we're gonna keep using the `setThresholds` as a way to make the user wait for calibration, we could
1. make the method completely block execution until calibration is complete. (Though I'm not sure how ROS handles action servers that block forever.)
2. only start the action server once calibration is complete, so that `waitForServer` takes care of that.

Reply:
If you like one of these ideas, I can implement them easily, otherwise I'd prefer creating an issue for that. Furthermore, it affects `MoveUntilTouchController` (without `Topic`) as well and we probably should also look at who uses/used that and how they dealt with that.

Reply:
My understanding is that `MoveUntilTouchController` hasn't been heavily tested on real hardware, but @mkoval or @ClintLiddick would know for sure.

(2) seems like a somewhat clean fix. Sounds fine to me, if it doesn't break anything 😅